### DISCUSSION TITLE
change worker_consistency to eventual

### DISCUSSION THREAD

**Discussion by hbagdi** [Kong/kong#7191] [Ideas and feature requests]
### Summary

`eventual` seems to be the right and sane default and we should switch to it as soon as possible. We keep running into it with customers with larger deployments.

**Comment by hbagdi** [comment]
xref https://github.com/Kong/charts/issues/228

**Comment by bungle** [comment]
@hbagdi, I agree, I want to propose deprecation of that setting and get rid of it with `3.0` (which would make it the default and the only option).

**Comment by hbagdi** [comment]
> I want to propose deprecation of that setting and get rid of it with 3.0 (which would make it the default and the only option).

+1 to that if that path is available

**Comment by scrudge** [comment]
We're trying to set this in our KIC deployment using environmental variables on the container.  We've tried KONG_WORKER_CONSISTENCY and KONG_NGINX_WORKER_CONSISTENCY  but neither seems to work.

**Comment by hbagdi** [comment]
Please make sure you are setting it on the kong container of the pod and not the controller container.

**Comment by scrudge** [comment]
I asked the same question.  We will validate, but I believe that we are putting them on the correct container.  

To validate that the settings are in effect, we are logging into the proxy container and looking for worker_consistency in the /usr/local/kong/nginx-kong.conf.  Is that the correct way to validate?

**Comment by hbagdi** [comment]
Umm, no.
You won't see any difference in that file.
Set KONG_WORKER_CONSISTENCY=eventual as an environment variable and you should be good.

https://docs.konghq.com/gateway-oss/2.4.x/configuration/#worker_consistency

**Comment by scrudge** [comment]
Ok, thanks.
Is there anyway way we can validate the the setting is changed?

**Comment by hbagdi** [comment]
`/usr/local/kong/.kong_env` should have that property set.
